scripts/tasks/preprocess-api-history.ts

```diff
@@ -0,0 +1,232 @@
+    (node: Node) =>
+      isHtml(node) &&
+      (node as LiteralString).value.includes('```') &&
+      (node as LiteralString).value.toLowerCase().includes('yaml') &&
+      (node as LiteralString).value.toLowerCase().includes('history'),
+    (node) => {
+      if (!isHtml(node)) return; // Should never happen, but TypeScript needs this check
+      codeBlocks.push(node);
+    }
+  );
+
+  return codeBlocks;
+}
+
+function findValidApiHistoryBlocks(
+  possibleHistoryBlocks: Html[],
+  filePath: string,
+  validateAgainstSchema: ValidateFunction<ApiHistory>
+) {
+  const validHistoryBlocks: Html[] = [];
+
+  // All of the validation logic copied from here: <https://github.com/electron/lint-roller/blob/bac245478ba69017c2a82e8ba1b2884a37647494/bin/lint-markdown-api-history.ts#L117-L176>
+  for (const possibleHistoryBlock of possibleHistoryBlocks) {
+    const {
+      children: [htmlComment],
+    } = fromHtml(possibleHistoryBlock.value);
+
+    if (htmlComment == null) {
+      logger.warn(
+        `Error parsing possible history block (found null/undefined htmlComment) in ${logger.green(
+          filePath
+        )}`
+      );
+      continue;
+    }
+
+    if (htmlComment.type !== 'comment') {
+      logger.warn(
+        `Possible API History block is not in a HTML comment (${logger.green(
+          filePath
+        )})`
+      );
+      continue;
+    }
+
+    const {
+      children: [codeBlock],
+    } = fromMarkdown(htmlComment.value);
+
+    if (codeBlock == null) {
+      logger.warn(
+        `Error parsing possible history block (found null/undefined codeBlock) in ${logger.green(
+          filePath
+        )}`
+      );
+      continue;
+    }
+
+    if (
+      codeBlock.type !== 'code' ||
+      codeBlock.lang?.toLowerCase() !== 'yaml' ||
+      codeBlock.meta?.trim() !== 'history'
+    ) {
+      logger.warn(
+        `Error parsing possible history block (codeBlock wasn't code, yaml, or history) in ${logger.green(
+          filePath
+        )}`
+      );
+      continue;
+    }
+
+    let unsafeHistory = null;
+
+    try {
+      unsafeHistory = parseYaml(codeBlock.value);
+    } catch (error) {
+      logger.warn(
+        `Error parsing YAML in possible history block (${logger.green(
+          filePath
+        )})`
+      );
+      continue;
+    }
+
+    const isValid = validateAgainstSchema(unsafeHistory);
+
+    if (!isValid) {
+      logger.warn(
+        `Error validating YAML in possible history block (${logger.green(
+          filePath
+        )})`
+      );
+      continue;
+    }
+
+    validHistoryBlocks.push(possibleHistoryBlock);
+  }
+
+  return validHistoryBlocks;
+}
+
+export const preprocessApiHistory = async (startPath: string) => {
+  const schema = path.resolve(startPath, 'api-history.schema.json');
+
+  let validateAgainstSchema: ValidateFunction<ApiHistory> | null = null;
+
+  try {
+    const ajv = new Ajv();
+    const ApiHistorySchemaFile = await readFile(schema, { encoding: 'utf-8' });
```

Comment:
Maintain a consistent code style.

```suggestion
    const ApiHistorySchemaFile = await readFile(schema, 'utf-8');
```